### DISCUSSION TITLE
Problems with RLS

### DISCUSSION THREAD

**Discussion by MarMedina95** [supabase/supabase#29099] [Questions]
Hi
I want to create a RLS in my table "estadisticas", but I want that this RLS is apply to the users that have the role of "ingenieria", and these user cannot see the info of this table that in the column of type has this info: 
- 'instalaciones_gpon_por_semana_tecnico'
-  'instalaciones_gpon_por_tecnicos_por_dia'
- 'instalaciones_gpon_por_dia'.

But the rest of the info in that table the users with that role could see it

This is the function and the RLS that I have done:

```
CREATE OR REPLACE FUNCTION get_user_roles(user_id UUID)
RETURNS JSONB AS $$
DECLARE
    roles JSONB;
BEGIN
    SELECT raw_user_meta_data->'roles' INTO roles
    FROM auth.users
    WHERE id = user_id;
    RETURN COALESCE(roles, '[]'::jsonb);  -- Asegurarte de que nunca devuelva NULL
END;
$$ LANGUAGE plpgsql;

CREATE POLICY "ingenieria_no_acceso" ON public.estadisticas AS RESTRICTIVE FOR
SELECT
  TO authenticated USING (
    get_user_roles (auth.uid ()) @> '["ingenieria"]'::jsonb
    AND tipo IN (
      'instalaciones_gpon_por_semana_tecnico',
      'instalaciones_gpon_por_tecnicos_por_dia',
      'instalaciones_gpon_por_dia'
    )
  );
```

But It is not working, could someone please explain me where is the error ? please

**Comment by GaryAustin1** [comment]
Possibly you can't access auth.users unless you have a security definer function.

Also you should never use raw_user_meta_data for a secure claim like role.  A user can change the metadata at anytime they are signed in by calling updateUser even directly with the REST API not using your code.

Some have used raw_app_meta_data for such a claim or a separate roles table.    Supabase also now has this method https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac?queryGroups=language&language=plpgsql
